### DISCUSSION TITLE
Why does Vercel's TZ env value have a colon in it?  `:UTC`

### DISCUSSION THREAD

**Discussion by cevaris** [vercel/community#913] [Help] [answered]
Good day, 

TZ is the environment variable that modifies Node's default timezone when calling `new Date()`. 
Per docs and enforced by UI, TZ value cannot be altered externally by Vercel customers. TZ is "reserved".

I have sanity check to make sure my code base is always running on TZ=UTC, so I can depend on that timezone throughout the code. 

Here is my code check which is executed at static build time... 
```
// NOTE: TZ is not used anywhere but is used for requring the TZ to be present.
// NOTE: Dont change TZ to be anything other than UTC, else new Date() will have the wronge time zone.
const TZ = required("TZ");
if (TZ !== 'UTC') {
    throw new Error(`TZ must be set to UTC, but is set to ${TZ}`);
}
```

This sanity check value fails with the following error when pushed to vercel builds...
```
> Build error occurred
Error: TZ must be set to UTC, but is set to :UTC
    at Object.9122 (/vercel/path0/.next/server/chunks/289.js:264:11)
    at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
    at Object.4091 (/vercel/path0/.next/server/chunks/289.js:156:14)
    at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
    at Object.5289 (/vercel/path0/.next/server/chunks/289.js:17:66)
    at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
    at Object.6251 (/vercel/path0/.next/server/pages/secret-sitemap.xml.js:16:87)
    at __webpack_exec__ (/vercel/path0/.next/server/pages/secret-sitemap.xml.js:141:39)
    at /vercel/path0/.next/server/pages/secret-sitemap.xml.js:142:74 {
  type: 'Error'
}
error Command failed with exit code 1.
``` 

Note, it seems Vercel's TZ value has an extra `:`/colon in front, `:UTC`. Is that intended? 

If it helps the internal team, here is an instance of my sanity check failing
https://vercel.com/acardenas/product-arc/DPdfscZBaet3WTXDAckMPcJLXvRo

**Comment by cevaris** [comment]
Indeed, altering my sanity check to the following makes my build pass..

```
const TZ = required("TZ");
if (TZ !== 'UTC' && TZ !== ':UTC') {
    throw new Error(`TZ must be set to UTC, but is set to ${TZ}`);
}
```

So seemingly Vercel introduced a colon in the TZ value. 
I am guessing that still works as intended, but scary, nonetheless. 

Passing build
https://vercel.com/acardenas/product-arc/F3gL7Zx5VDA1QY1v9LsHTbpXN5b2

**Comment by gudmundur** [comment, marked as answer]
Thanks for raising this issue with us. I've been looking into this, and I do see that we set `TZ=:UTC`. This is a change from a couple of years, so I don't think it's been causing any obvious issues for customers. After reading [the `libc` docs for `TZ`](https://www.gnu.org/software/libc/manual/html_node/TZ-Variable.html), this particular value seems non-standard. Normally when setting time zones, I've seen `/etc/localtime` being modified, usually by setting a symlink to the correct time zone information file, which in this case is `/usr/share/zoneinfo/UTC`. I did try unsetting `TZ` in our development environment and it didn't seem to cause any regressions.

If you can live with the updated code as you described, then I'd recommend that. As for moving forward, I will bring this up with the team, and see whether we change this default.

**Comment by cevaris** [comment]
Thanks for confirming. My hack seems to be working, so def fine leaving as is. Thanks.
